pennylane_lightning_gpu/lightning_gpu.py

```diff
@@ -152,9 +233,216 @@ def syncH2D(self, use_async=False):
 
     def syncD2H(self, use_async=False):
         """Explicitly synchronize GPU data to CPU"""
+        if len(self._state) != 2**self.num_wires:
+            state = np.zeros(2**self.num_wires, dtype=np.complex128)
+            state = self._asarray(state, dtype=self.C_DTYPE)
+            self._state = self._reshape(state, [2] * self.num_wires)
+
         self._gpu_state.DeviceToHost(self._state.ravel(order="C"), use_async)
         self._pre_rotated_state = self._state
 
+    @property
+    def state(self):
+        # Flattening the state.
+        shape = (1 << self.num_wires,)
+        self.syncD2H()
+        return self._reshape(self._pre_rotated_state, shape)
```

Comment:
As with here, we don't want a `self._state` or `self.state` property on the host.

Reply:
If we keep the `_state`, which can be set as an empty np.array, attribute here, it will make the testing simpler. We need to discuss about it.